### PR TITLE
Add Issues/Notes

### PR DESCRIPTION
**Pull request by OlegMoshkovich**

### CODE REVIEW

**Comment by OlegMoshkovich** [comment]
Morning! Addressed the store reset and the mobile drawer issues.

File: src/Components/IssuesControl.jsx [hunk 1]
```diff
@@ -1,181 +1,270 @@
...
-import DialogActions from '@mui/material/DialogActions'
-import DialogTitle from '@mui/material/DialogTitle'
-import DialogContent from '@mui/material/DialogContent'
-import MuiDialog from '@mui/material/Dialog'
-import Typography from '@mui/material/Typography'
-import Slide from '@mui/material/Slide'
-import {makeStyles} from '@mui/styles'
-import {ControlButton, TooltipIconButton} from './Buttons'
+import React, {useEffect} from 'react'
 import debug from '../utils/debug'
```
Comment (OlegMoshkovich) [inline src/Components/IssuesControl.jsx:2]: @pablo-mayrgundter can you please look over this file.

File: src/Components/IssuesControl.jsx [hunk 2]
```diff
@@ -1,181 +1,270 @@
...
+  const selectedIssueId = useStore((state) => state.selectedIssueId)
+  const issuesStore = useStore((state) => state.issues)
+  const setIssuesStore = useStore((state) => state.setIssues)
+  const comments = useStore((state) => state.comments)
+  const setComments = useStore((state) => state.setComments)
+  const filteredIssue = selectedIssueId ? issuesStore.filter((issue) => issue.id ===selectedIssueId)[0] :null
 
-/** The prefix to use for issue id in the URL hash. */
-export const ISSUE_PREFIX = 'i'
+  useEffect(()=>{
```
Comment (OlegMoshkovich) [inline src/Components/IssuesControl.jsx:127]: @pablo-mayrgundter please look over this

File: src/Components/CameraControl.jsx [hunk 3]
```diff
@@ -19,16 +20,19 @@ import {roundCoord} from '../utils/math'
  * URL hash and sets the camera position, as well as adds a hash
  * listener to do the same whenever the hash changes.
  *
- * @param {Object} viewer The IFC viewer, which contains the
+ * @param {Object} v The IFC viewer, which contains the
```
Comment (pablo-mayrgundter) [inline src/Components/CameraControl.jsx:23]: should still be 'viewer'

File: src/Components/CameraControl.jsx [hunk 4]
```diff
@@ -116,7 +127,7 @@ export function parseHashParams(encodedParams) {
 }
 
 
-/** @return {boolean} True iff the camera hash params are present. */
+/** @return {boolean} True if the camera hash params are present. */
```
Comment (pablo-mayrgundter) [inline src/Components/CameraControl.jsx:130]: Should be 'iff'.  See https://en.wikipedia.org/wiki/If_and_only_if

File: src/Components/CameraControl.jsx [hunk 5]
```diff
@@ -128,13 +139,15 @@ export function hasValidUrlParams() {
...
 /**
  * Adds camera coords to url.
- * @param {Object} viewer IFCjs viewer that contains a camera.
+ * @param {Object} cameraControls obtained from the viewer
  */
-export function addCameraUrlParams(viewer) {
-  const camera = viewer.IFC.context.ifcCamera
-  const position = camera.cameraControls.getPosition()
+export function addCameraUrlParams(cameraControls) {
+  if (!cameraControls ) {
```
Comment (pablo-mayrgundter) [inline src/Components/CameraControl.jsx:145]: This seems like an error.  Is there a valid case where this method is called with null/undefined?  If so pls document if not, throw error

File: src/Components/IssueCard.jsx [hunk 6]
```diff
@@ -0,0 +1,362 @@
+import React, {useState, useEffect} from 'react'
+import Paper from '@mui/material/Paper'
+import {makeStyles} from '@mui/styles'
+import Select from '../assets/2D_Icons/Select.svg'
```
Comment (pablo-mayrgundter) [inline src/Components/IssueCard.jsx:4]: Please sort svgs to bottom

File: src/Components/IssueCard.jsx [hunk 7]
```diff
@@ -0,0 +1,362 @@
...
+  const setSelectedCommentIndex = useStore((state) => state.setSelectedCommentIndex)
+  const setSelectedIssueId = useStore((state) => state.setSelectedIssueId)
+  const setSnackMessage = useStore((state) => state.setSnackMessage)
+  const selected = selectedIssueId === id
+  const textOverflow = body.length > 80
+  const isImage = imageURL.length !=0
+  const isMobile = useIsMobile()
+  const classes = useStyles({expandText: expandText, select: selected, expandImage: expandImage})
+
+  useEffect(()=>{
```
Comment (pablo-mayrgundter) [inline src/Components/IssueCard.jsx:48]: Please format the arrow functions in this file

File: src/Components/IssueCard.jsx [hunk 8]
```diff
@@ -0,0 +1,362 @@
...
+import {ISSUE_PREFIX} from './IssuesControl'
+import {addHashParams} from '../utils/location'
+import {setCameraFromEncodedPosition, addCameraUrlParams, removeCameraUrlParams} from './CameraControl'
+import {useIsMobile} from './Hooks'
+
+
+/**
+ * Issue card
+ * @param {string} title The comment body
+ * @param {string} contetne The comment title, optional
```
Comment (pablo-mayrgundter) [inline src/Components/IssueCard.jsx:18]: these params are out of sync

File: src/Components/IssueCard.jsx [hunk 9]
```diff
@@ -0,0 +1,362 @@
...
+ * @return {Object} React component
+ */
+export default function IssueCard({
```
Comment (pablo-mayrgundter) [inline src/Components/IssueCard.jsx:21]: So, it's not clear to me what's required or not here.  Anything that's actually required should have an assertDefined here before continuing to pass values downwards.

File: src/Components/IssuesControl.jsx [hunk 10]
```diff
@@ -1,181 +1,270 @@
...
 } from '../utils/location'
+import {makeStyles, useTheme} from '@mui/styles'
 import {getIssue, getComment} from '../utils/GitHub'
-import CommentIcon from '../assets/2D_Icons/Comment.svg'
-import NavPrevIcon from '../assets/2D_Icons/NavPrev.svg'
-import NavNextIcon from '../assets/2D_Icons/NavNext.svg'
+import Paper from '@mui/material/Paper'
+import {getIssues, getComments} from '../utils/GitHub'
+import IssueCard from './IssueCard'
+import Next from '../assets/2D_Icons/NavNext.svg'
```
Comment (pablo-mayrgundter) [inline src/Components/IssuesControl.jsx:11]: sort svgs to bottom

File: src/Components/IssuesControl.jsx [hunk 11]
```diff
@@ -1,181 +1,270 @@
...
+    }
+    fetchIssues()
+  }, [setIssuesStore])
 
-// exported for testing only
-const regex = new RegExp(`${ISSUE_PREFIX}:(\\d+)(?::(\\d+))?`)
 
+  useEffect(()=>{
+    const fetchComments = async (selectedIssue) => {
+      console.log('comments - fetch loop', selectedIssueId)
```
Comment (pablo-mayrgundter) [inline src/Components/IssuesControl.jsx:171]: remove

File: src/Components/OperationsGroup.jsx [hunk 12]
```diff
@@ -1,14 +1,15 @@
...
-import IssuesControl from './IssuesControl'
 import ShareControl from './ShareControl'
 import ShortcutsControl from './ShortcutsControl'
 import {TooltipIconButton} from './Buttons'
 import CutPlaneIcon from '../assets/2D_Icons/CutPlane.svg'
 import ClearIcon from '../assets/2D_Icons/Clear.svg'
+import MarkupIcon from '../assets/2D_Icons/Markup.svg'
 import ListIcon from '../assets/2D_Icons/List.svg'
 import {useIsMobile} from './Hooks'
+// import {removeCameraUrlParams} from './CameraControl'
```
Comment (pablo-mayrgundter) [inline src/Components/OperationsGroup.jsx:12]: remove

File: src/Components/OperationsGroup.jsx [hunk 13]
```diff
@@ -22,38 +23,45 @@ import useStore from '../store/useStore'
...
   return (
     <div className={classes.container}>
-      <div className={classes.shareAndIssues}>
+      <div className={classes.topGroup}>
         <ShareControl viewer={viewer}/>
-        <IssuesControl viewer={viewer}/>
-
-
+        <TooltipIconButton
+          title='Notes'
```
Comment (pablo-mayrgundter) [inline src/Components/OperationsGroup.jsx:42]: format spaces around =, and below a few places

File: src/Components/OperationsGroup.jsx [hunk 14]
```diff
@@ -22,38 +23,45 @@ import useStore from '../store/useStore'
...
+          icon ={<MarkupIcon/>}
+          onClick = {
```
Comment (pablo-mayrgundter) [inline src/Components/OperationsGroup.jsx:44]: one line, and below

File: src/utils/GitHub.js [hunk 15]
```diff
@@ -1,7 +1,7 @@
 import {Octokit} from '@octokit/rest'
 import PkgJson from '../../package.json'
 import debug from './debug'
-import {isRunningLocally} from './network'
+// import {isRunningLocally} from './network'
```
Comment (pablo-mayrgundter) [inline src/utils/GitHub.js:4]: revert

File: src/utils/GitHub.js [hunk 16]
```diff
@@ -17,6 +17,37 @@ export async function getIssue(issueId) {
...
+  const comments = await getGitHub(
+      'issues/{issue_number}/comments',
+      {
+        issue_number: issueId,
+      })
+  debug().log('GitHub: comments: ', comments)
+  if (comments && comments.data && comments.data.length > 0) {
+    return comments.data
+  } else {
+    console.warn('Empty comments!')
```
Comment (pablo-mayrgundter) [inline src/utils/GitHub.js:46]: debug()

File: src/utils/GitHub.js [hunk 17]
```diff
@@ -65,75 +96,121 @@ async function getGitHub(path, args) {
...
-workspaces, open standards and open source code the most powerful way
-to work. Cooperation is the unfair advantage.\r \`\`\`\r
-url=//${window.location.host}/share/v/p/index.ifc#i:8::c:-144.36,14.11,147.82,-40.42,17.84,-2.28\r
-\`\`\`\r
-`,
-    },
-  ],
-}
-
+// class MockOctokit {
```
Comment (pablo-mayrgundter) [inline src/utils/GitHub.js:99]: Remove the rest

**Review by pablo-mayrgundter — CHANGES_REQUESTED** [review]

File: src/Components/CameraControl.jsx (hunk 3 above)
Comment (OlegMoshkovich) [reply src/Components/CameraControl.jsx:23]: ok

**Comment by OlegMoshkovich** [comment]
@pablo-mayrgundter PTAL

**Review by OlegMoshkovich — COMMENTED** [review]
fixed spacing with eslint rules
@pablo-mayrgundter PTAL

File: src/Components/IssueCard.jsx [hunk 18]
```diff
@@ -0,0 +1,383 @@
...
+ * Issue card
+ * @param {number} id issue id
+ * @param {number} index issue index
+ * @param {string} username username of the issue author
+ * @param {string} title issue title
+ * @param {string} avatarUrl user avatarUrl
+ * @param {string} body issue body
+ * @param {string} imageUrl issue image
+ * @param {string} date issue date
+ * @param {string} url full url attached to GH issue with camera position
```
Comment (pablo-mayrgundter) [inline src/Components/IssueCard.jsx:26]: embeddedUrl

File: src/Components/IssueCard.jsx [hunk 19]
```diff
@@ -0,0 +1,383 @@
...
+  )
+}
+
+const CardImage = ({imageUrl, onClickImage, expandImage}) => {
+  const classes = useStyles({expandImage: expandImage})
+  return (
+    <div className = {classes.imageContainer}
+      onClick = {onClickImage}
+      role = 'button'
+      tabIndex={0}
```
Comment (pablo-mayrgundter) [inline src/Components/IssueCard.jsx:181]: looks like you still need to activate eslint spaces checking

File: src/Components/IssueCard.test.jsx [hunk 20]
```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import {render, screen, fireEvent} from '@testing-library/react'
+import {MockRoutes} from '../BaseRoutesMock.test'
+import IssueCard from './IssueCard'
+
+
+test('IssueCard', () => {
+  const id = 123
+  const index = 123
+  render(<MockRoutes contentElt={<IssueCard id={id} index = {index} title = "new_title" />}/>)
```
Comment (pablo-mayrgundter) [inline src/Components/IssueCard.test.jsx:10]: same, spaces

File: src/Components/IssueCard.test.jsx [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+  fireEvent.click(selectIssueButton)
+  expect(selectIssueButton).not.toBeInTheDocument()
+})
+
+
+test('Camera Position control', () => {
+  const id = 123
+  const index = 123
+  const rendered = render(<MockRoutes contentElt={<IssueCard id={id} index = {index}
+    embeddedUrl = " http://localhost:8080/share/v/p/index.ifc#c:-141.9,72.88,21.66,-43.48,15.73,-4.34" />}/>)
```
Comment (pablo-mayrgundter) [inline src/Components/IssueCard.test.jsx:37]: nice!

File: src/Components/IssueControl.test.jsx [hunk 22]
```diff
@@ -0,0 +1,110 @@
...
+import useStore from '../store/useStore'
+
+
+const MOCK_ISSUES = [
+  {
+    embeddedUrl: 'url = http://localhost:8080/share/v/p/index.ifc#c:-141.9,72.88,21.66,-43.48,15.73,-4.34',
+    index: 0,
+    id: 10,
+    number: 1,
+    title: 'TEST_ISSUE_TITLE_1',
```
Comment (pablo-mayrgundter) [inline src/Components/IssueControl.test.jsx:15]: can you use a different title and body throughout?  I suggest 2 sentences with a few words each.  These will live with us for a while and will help us test preview, etc. later on

File: src/Components/OperationsGroup.jsx [hunk 23]
```diff
@@ -22,45 +22,48 @@ import useStore from '../store/useStore'
...
+            onClick = {() => viewer.clipper.createPlane()}
+            icon = {<CutPlaneIcon/>}/> :
           null
         }
-        <TooltipIconButton title="Clear selection" onClick={unSelectItem} icon={<ClearIcon/>}/>
+        <TooltipIconButton title="Clear selection" onClick = {unSelectItem} icon = {<ClearIcon/>}/>
         <ShortcutsControl/>
       </div>
       {/* Invisible */}
-      <CameraControl viewer={viewer}/>
```
Comment (pablo-mayrgundter) [inline src/Components/OperationsGroup.jsx:63]: good to see consistent style, but i think to minimize change for now we should use no spaces, as in discussion on discorc

File: src/Components/SideDrawer.jsx [hunk 24]
```diff
@@ -4,50 +4,64 @@ import {makeStyles} from '@mui/styles'
...
 
   return (
     <>
-      {
-        isMobile && isDrawerOpen ?
+      {isMobile && isDrawerOpen ?
         <MobileDrawer
-          content={<PropertiesPanel/>}/> :
+          content = {
+            <div className={classes.content}>
```
Comment (pablo-mayrgundter) [inline src/Components/SideDrawer.jsx:36]: spaces in whole file

File: src/utils/GitHub.js (hunk 15 above)
Comment (pablo-mayrgundter) [reply src/utils/GitHub.js:4]: I do think we want this check for running local.. the problem is that if you use live comments from GitHub when developing you can quickly hit per-user quota limits.  I added this to use the test comments when developing and then flip them when I ready to test with more realistic data.

File: src/utils/GitHub.js [hunk 25]
```diff
@@ -17,6 +16,37 @@ export async function getIssue(issueId) {
...
+ */
+export async function getIssues() {
+  const issues = await getGitHub('issues')
+  debug().log('GitHub: issue: ', issues)
+  return issues
+}
+
+
+/**
+ * The comments should have the following structure:
```
Comment (pablo-mayrgundter) [inline src/utils/GitHub.js:31]: Stale comment?

File: src/utils/GitHub.js [hunk 26]
```diff
@@ -17,6 +16,37 @@ export async function getIssue(issueId) {
...
+ * @param {Number} issueId
+ * @return {Array} The comments array.
+ */
+export async function getComments(issueId) {
+  const comments = await getGitHub(
+      'issues/{issue_number}/comments',
+      {
```
Comment (pablo-mayrgundter) [inline src/utils/GitHub.js:38]: yep, that's right

File: src/utils/GitHub.js [hunk 27]
```diff
@@ -134,6 +164,9 @@ url=//${window.location.host}/share/v/p/index.ifc#i:8::c:-144.36,14.11,147.82,-4
 
 // All direct uses of octokit should be private to this file to
 // ensure we setup mocks for local use and unit testing.
-const octokit = isRunningLocally() ? new MockOctokit() : new Octokit({
+// const octokit = isRunningLocally() ? new MockOctokit() : new Octokit({
```
Comment (pablo-mayrgundter) [inline src/utils/GitHub.js:167]: please restore unless you found a good reason not to

**Comment by pablo-mayrgundter** [comment]
Code is looking good.  But comment link doesn't open the issue panel!

https://deploy-preview-235--bldrs-share.netlify.app/share/v/p/index.ifc/89/112/139/154/265#c:22.29,49.57,82.32,29.19,6.12,-9.25::i:1257156364

This brings me to the correct item in the scene, but the issue panel isn't open.  When i do open it, the issue isn't selected.

File: src/Components/SideDrawer.jsx [hunk 28]
```diff
@@ -4,50 +4,110 @@ import {makeStyles} from '@mui/styles'
...
+}
 
+
+/**
+ * SideDrawerWrapper is the container for the SideDrawer component.
+ * it is loaded into the CadView, connected to the store and passes the props to the sideDrawer.
+ * It makes it is possible to test Side Drawer outside of the cad view.
+ * @return {Object} SideDrawer react component
+ */
+export default function SideDrawerWrapper() {
```
Comment (OlegMoshkovich) [inline src/Components/SideDrawer.jsx:80]: @pablo-mayrgundter added the wrapper

File: src/Components/SideDrawer.test.jsx [hunk 29]
```diff
@@ -0,0 +1,48 @@
...
+    result.current.setSelectedElement(MOCK_SELECTED_ELEMENT)
+    result.current.toggleIsPropertiesOn()
+    result.current.openDrawer()
+  })
+  render(<MockRoutes contentElt={<SideDrawerWrapper/>}/>)
+  expect(screen.getByText('Properties')).toBeInTheDocument()
+})
+
+
+test('side drawer - issues id in url', () => {
```
Comment (OlegMoshkovich) [inline src/Components/SideDrawer.test.jsx:36]: @pablo-mayrgundter side drawer test

**Comment by OlegMoshkovich** [comment]
@pablo-mayrgundter PTAL

File: src/Containers/CadView.jsx [hunk 30]
```diff
@@ -228,16 +226,6 @@ export default function CadView({
     setupLookupAndParentLinks(rootElt, elementsById)
     setDoubleClickListener()
     initSearch(model, rootElt)
-    // The spatial structure doesn't contain properties.  NavTree uses
```
Comment (pablo-mayrgundter) [inline src/Containers/CadView.jsx:231]: Looks like a bad merge.. this is live in prod and should be included.

File: src/Components/OperationsGroup.jsx [hunk 31]
```diff
@@ -22,38 +22,43 @@ import useStore from '../store/useStore'
...
 
-  const toggle = () => {
-    isDrawerOpen ? closeDrawer() : openDrawer(),
-    toggleIsPropertiesOn()
+  const toggle = (panel) => {
+    openDrawer()
+    panel === 'Properties' ? toggleIsPropertiesOn() : null
+    panel === 'Notes' ? toggleIsCommentsOn() : null
   }
+  console.log('selected element', selectedElement)
```
Comment (pablo-mayrgundter) [inline src/Components/OperationsGroup.jsx:36]: remove

File: src/Components/SideDrawer.test.jsx [hunk 32]
```diff
@@ -0,0 +1,48 @@
...
+  const {result} = renderHook(() => useStore((state) => state))
+  const extractedCommentId = '1257156364'
+  act(() => {
+    result.current.setSelectedIssueId(Number(extractedCommentId))
```
Comment (pablo-mayrgundter) [inline src/Components/SideDrawer.test.jsx:40]: This simulates the after-effect of an issue in the URL.  I think IssuesControl   We also need to ensure the trip from the the URL to here.

**Review by pablo-mayrgundter — CHANGES_REQUESTED** [review]
Let's talk about the last unit test to exercise the code getting the hash from useLocation through to IssueCard

File: src/utils/GitHub.js [hunk 33]
```diff
@@ -17,6 +16,38 @@ export async function getIssue(issueId) {
...
+ * @return {Array} The comments array.
+ */
+export async function getComments(issueId) {
+  const comments = await getGitHub(
+      'issues/{issue_number}/comments',
+      {
+        issue_number: issueId,
+      })
+  debug().log('GitHub: comments: ', comments)
+  console.log('comments', comments)
```
[the remark this answers is not in the document]
Comment (OlegMoshkovich) [reply src/utils/GitHub.js]: all of the console logs are outdated. - not sure why it is showing up here.

File: src/Components/SideDrawer.test.jsx (hunk 29 above)
Comment (pablo-mayrgundter) [reply src/Components/SideDrawer.test.jsx:42]: Thanks!

File: src/Components/SideDrawer.jsx (hunk 28 above)
Comment (pablo-mayrgundter) [reply src/Components/SideDrawer.jsx:80]: ok

File: src/Components/IssuesControl.jsx [hunk 34]
```diff
@@ -1,181 +1,275 @@
...
+  const issues = useStore((state) => state.issues)
+  const setIssues = useStore((state) => state.setIssues)
+  const comments = useStore((state) => state.comments)
+  const setComments = useStore((state) => state.setComments)
+  const filteredIssue = selectedIssueId ? issues.filter((issue) => issue.id === selectedIssueId)[0] : null
 
-/** The prefix to use for issue id in the URL hash. */
-export const ISSUE_PREFIX = 'i'
+  useEffect(() => {
+    const fetchIssues = async () => {
```
Comment (pablo-mayrgundter) [inline src/Components/IssuesControl.jsx:126]: Please add:
```
// TODO(pablo): refactor issues fetch logic into Issues.js
```

**Review by pablo-mayrgundter — APPROVED** [review]

**Comment by pablo-mayrgundter** [comment]
Please add the one TODO comment I noted, then all good!

**Comment by pablo-mayrgundter** [comment]
Actually, I want to base some changes off of this so will merge now and add the todo.

**Pull request merged**